lustre/utils/lfs.c

```diff
@@ -1015,6 +1018,69 @@ static int parse_targets(__u32 *osts, int size, int offset, char *arg)
 	return rc < 0 ? rc : nr;
 }
 
+#define LFS_PRIORITY		(LOG_DAEMON | LOG_INFO)
+#define LFS_MSG_TAG		"lfs_arguments"
+#define LFS_JOBID_VAR		"SLURM_JOB_ID"
+static void log_to_syslog(const char *in_path)
+{
+	char rp[PATH_MAX];
+	char *have_rp;
+	struct utsname uts;
+	int have_uts;
+	char *jobid;
+
+	jobid = getenv(LFS_JOBID_VAR);
+	have_uts = !uname(&uts);
+
+	memset(rp, 0, sizeof(rp));
+
+	/*
+	 * lfs setstripe must be called on a path that either
+	 * (a) refers to a directory, which must exist
+	 * (b) refers to a file, which must not exist, in a directory, which
+	 *     does exist.
+	 *
+	 * If in_path doesn't yet exist, realpath fails with ENOENT.  In that
+	 * case, rp points to the "prefix of the path that does not exist"
+	 * (which is the full path including file name). This is a GNU
+	 * extension, see realpath(3).
+	 *
+	 * In that case, realpath() has done everything to that "prefix" that it
+	 * normally does - resolve symlinks, make absolute, etc.
+	 *
+	 * If the argument path refers to a directory, it must exist, so
+	 * realpath() succeeds.  Either way, realpath() does the right thing.
+	 *
+	 * If the user has provided a path in which one or more directories does
+	 * not exist, lfs setstripe will fail and we do not need report it the
+	 * path at all.  We attempt to detect this by comparing suffixes.  This
+	 * is not always correct, but likely good enough since we are using this
+	 * only to identify good places to test PFL.
+	 */
+
+	errno = 0;
+	have_rp = realpath(in_path, rp);
+	if (!have_rp && errno == ENOENT) {
+		char *in_suffix;
+		char *rp_suffix;
+		char *in_dupe;
+
+		in_dupe = strdup(in_path);
+		in_suffix = basename(in_dupe);
+		free(in_dupe);
```

Comment:
I think that the free() could be done at the end of the function. The man page for basename(3) says not to free the input to basename before it's return value is done being used.

Reply:
yep, that was a mistake.  Fixed.

Reply:
I moved the free to the end of the block, rather than the end of the function.  in_dupe last used by the strcmp() a couple of lines down.  The free() is now after that.